src/hip_zfp/variable.h

```diff
@@ -192,12 +192,13 @@ namespace hipZFP
         for (int i = 0; i < nstreams_chunk; i += grid_stride)
         {
             bool valid_stream = my_stream + i < nstreams_chunk;
-            unsigned long long offset0 = offsets[first_bitstream_block + i];
+            unsigned long long offset0 = 0;
             unsigned long long offset = 0;
             uint length_bits = 0;
             uint add_padding = 0;
             if (valid_stream)
             {
+                offset0 = offsets[first_bitstream_block + i];
```

Comment:
I've not yet fully absorbed how this code functions, but I can see how this might cause a memory violation for certain data/chunk sizes.  It seems clear that the same fix should apply to the CUDA implementation.  I will go ahead and merge this and apply the fix to CUDA myself.  Thanks for tracking down this bug!